test/common/k8s/azureidentity/azureidentity.go

```diff
@@ -0,0 +1,191 @@
+package azureidentity
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"time"
+
+	aadpodid "github.com/Azure/aad-pod-identity/pkg/apis/aadpodidentity/v1"
+	"github.com/Azure/aad-pod-identity/test/common/util"
+	"github.com/pkg/errors"
+)
+
+// CreateOnAzure will create a user-assigned identity on Azure, assign 'Reader' role
+// to the identity and assign 'Managed Identity Operator' role to service principal
+func CreateOnAzure(subscriptionID, resourceGroup, azureClientID, name string) error {
+	cmd := exec.Command("az", "identity", "create", "-g", resourceGroup, "-n", name)
+	_, err := cmd.CombinedOutput()
+	if err != nil {
+		return errors.Wrap(err, "Failed to create a user-assigned identity on Azure")
+	}
+
+	// Assigning 'Reader' role to the identity
+	_, err = WaitOnReaderRoleAssignment(resourceGroup, name)
+	if err != nil {
+		return err
+	}
+
+	// Assign 'Managed Identity Operator' to Service Principal
+	cmd = exec.Command("az", "role", "assignment", "create", "--role", "Managed Identity Operator", "--assignee", azureClientID, "--scope", "/subscriptions/"+subscriptionID+"/resourcegroups/"+resourceGroup+"/providers/Microsoft.ManagedIdentity/userAssignedIdentities/"+name)
+	_, err = cmd.CombinedOutput()
+	if err != nil {
+		return errors.Wrap(err, "Failed to assign 'Managed Identity Operator' role to service principal on Azure")
+	}
+
+	return nil
+}
+
+// DeleteOnAzure will delete a given user-assigned identity on Azure
+func DeleteOnAzure(resourceGroup, name string) error {
+	cmd := exec.Command("az", "identity", "delete", "-g", resourceGroup, "-n", name)
+	_, err := cmd.CombinedOutput()
+	if err != nil {
+		return errors.Wrap(err, "Failed to delete the Azure identity from Azure")
+	}
+
+	return nil
+}
+
+// CreateOnCluster will create an Azure Identity on a Kubernetes cluster
+func CreateOnCluster(subscriptionID, resourceGroup, name, templateOutputPath string) error {
+	clientID, err := GetClientID(resourceGroup, name)
+	if err != nil {
+		return err
+	}
+
+	t, err := template.New("aadpodidentity.yaml").ParseFiles(path.Join("template", "aadpodidentity.yaml"))
+	if err != nil {
+		return errors.Wrap(err, "Failed to parse aadpodidentity.yaml")
+	}
+
+	deployFilePath := path.Join(templateOutputPath, name+".yaml")
+	deployFile, err := os.Create(deployFilePath)
+	if err != nil {
+		return errors.Wrap(err, "Failed to create a deployment file from aadpodidentity.yaml")
+	}
+	defer deployFile.Close()
+
+	deployData := struct {
+		SubscriptionID string
+		ResourceGroup  string
+		ClientID       string
+		Name           string
```

Comment:
nit: Can you please add here to the comment on how it ends up with the aadpodidentity .Name. Also other variables.